app/command/registration.js

```diff
@@ -46,19 +46,23 @@ class Registration {
 		};
 
 		function add( user ) {
-			registered.push( user );
+			// Add the player only if he/she is not registered already.
+			// We could return if user in in registered pool already but it could be abused to check if someone else is already registered.
+			if ( registered.indexOf( user ) === -1 ) {
+				registered.push( user );
```

Comment:
`registered` is last 20 registered users. The plan was to handle removing player from the previous match, not only the last one. It means that with this code user can not registered if he was registered in any of the last 5 matches.